crates/astria-sequencer/src/accounts/query.rs

```diff
@@ -215,3 +220,84 @@ async fn preprocess_request(
     };
     Ok((address, snapshot, height))
 }
+
+/// Custom deterministic sorting function for `AssetBalance` that sorts by balance in descending
+/// order and then by denom in ascending order. This function should never return `Ordering::Equal`,
+/// as calling `sort_unstable_by` on equal values has a non-deterministic result.
+fn compare_asset_balances(lhs: &AssetBalance, rhs: &AssetBalance) -> std::cmp::Ordering {
+    use std::cmp::Ordering;
+
+    match lhs.balance.cmp(&rhs.balance) {
+        // Denoms should never have the same display name
+        Ordering::Equal => lhs.denom.to_string().cmp(&rhs.denom.to_string()),
+        Ordering::Less => Ordering::Greater,
+        Ordering::Greater => Ordering::Less,
+    }
+}
+
+mod test {
```

Comment:
```suggestion
#[cfg(test)]
mod tests
```

Reply:
Reopened the issue: the convention is to call this `mod tests`, not `test` as per the suggestion.

Reply:
Sorry about that, should be fixed in 535c1c4